integrations/maya/src/main/java/org/dash/wallet/integrations/maya/ui/MayaCryptoCurrencyPickerFragment.kt

```diff
@@ -0,0 +1,162 @@
+ */
+
+package org.dash.wallet.integrations.maya.ui
+
+import android.os.Bundle
+import android.view.View
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import org.dash.wallet.common.ui.decorators.ListDividerDecorator
+import org.dash.wallet.common.ui.dialogs.AdaptiveDialog
+import org.dash.wallet.common.ui.radio_group.IconSelectMode
+import org.dash.wallet.common.ui.radio_group.IconifiedViewItem
+import org.dash.wallet.common.ui.recyclerview.IconifiedListAdapter
+import org.dash.wallet.common.ui.viewBinding
+import org.dash.wallet.common.util.GenericUtils
+import org.dash.wallet.common.util.observe
+import org.dash.wallet.integrations.maya.R
+import org.dash.wallet.integrations.maya.databinding.FragmentCurrencyPickerBinding
+import org.dash.wallet.integrations.maya.model.PoolInfo
+
+@AndroidEntryPoint
+class MayaCryptoCurrencyPickerFragment : Fragment(R.layout.fragment_currency_picker) {
+    private val binding by viewBinding(FragmentCurrencyPickerBinding::bind)
+    private val viewModel by viewModels<MayaViewModel>()
+    private var itemList = listOf<IconifiedViewItem>()
+    private lateinit var defaultItemMap: Map<String, IconifiedViewItem>
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.toolbar.setNavigationOnClickListener {
+            findNavController().popBackStack()
+        }
+
+        val adapter = IconifiedListAdapter() { item, index ->
+            viewModel.poolList.value.firstOrNull {
+                it.currencyCode == item.title
+            }?.let {
+                clickListener(it)
+            }
+        }
+
+        // val adapter1 = ListAdapter<IconifiedViewItem, RadioButtonViewHolder>(RadioGroupAdapter.DiffCallback())
+
+        val divider = ContextCompat.getDrawable(requireContext(), org.dash.wallet.common.R.drawable.list_divider)!!
+        val decorator = ListDividerDecorator(
+            divider,
+            showAfterLast = false,
+            marginStart = resources.getDimensionPixelOffset(org.dash.wallet.common.R.dimen.divider_margin_horizontal),
+            marginEnd = resources.getDimensionPixelOffset(org.dash.wallet.common.R.dimen.divider_margin_horizontal)
+        )
+        binding.contentList.addItemDecoration(decorator)
+        binding.contentList.adapter = adapter
+
+        // using this allows for translation of cryptocurrency names
+        defaultItemMap = mapOf(
+            "BTC.BTC" to IconifiedViewItem(
+                requireContext().getString(R.string.cryptocurrency_bitcoin_code),
+                requireContext().getString(R.string.cryptocurrency_bitcoin_network)
+                // R.drawable.ic_btc_logo
```

Comment:
originally the main point of this was to private a default list that could be displayed if the Maya servers were offline.  If its offline, then the service doesn't work.  The second thing is that it stores an association between the Maya asset and the name of the crypto currency.